### PR TITLE
Api integration controller tests

### PR DESCRIPTION
**Pull request by ffasitella**
### Summary

Adding tests for the OpenLibrary controller. 

### Trello card

https://trello.com/c/oYTpzd89/39-crear-test-para-el-endpoint-de-book-information

### CODE REVIEW

File: app/controllers/api/v1/open_library_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def show
         book = api_request
         return book_not_found if book.blank?
 
-        render json: book
+        render json: book, status: :ok
```
Comment (marcegl91) [inline app/controllers/api/v1/open_library_controller.rb]: This is not needed, the default response code for Rails is `:ok`

File: spec/controllers/api/v1/open_library_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        JSON.parse(
+          File.read('./spec/support/fixtures/open_library_service_response_success.json'),
+          symbolize_names: true
+        ).to_json
+      end
+
+      before do
+        stubbed_service = instance_double(OpenLibraryService)
+        allow(stubbed_service).to receive(:api_request)
+          .with(valid_isbn)
```
Comment (marcegl91) [inline spec/controllers/api/v1/open_library_controller_spec.rb]: You can set the stub with both `valid_isbn` and `wrong_isbn` outside of the context and you don't need to repeat the creation of this `stubbed_service`.

File: spec/controllers/api/v1/open_library_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      before do
+        stubbed_service = instance_double(OpenLibraryService)
+        allow(stubbed_service).to receive(:api_request)
+          .with(wrong_isbn)
+          .and_return(not_found_result)
+        allow(OpenLibraryService).to receive(:new).and_return(stubbed_service)
+        get :show, params: { isbn: wrong_isbn }
+      end
+
+      it 'returns an error message' do
```
Comment (marcegl91) [inline spec/controllers/api/v1/open_library_controller_spec.rb]: You can test also the HTTP code when the book is not found.

File: spec/controllers/api/v1/open_library_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      end
+    end
+
+    context 'When using an invalid ISBN' do
+      before do
+        get :show, params: { isbn: wrong_isbn }
+      end
+
+      it 'returns an error message' do
+        expect(response.body).to be_json_eql not_found_result
```
Comment (marcegl91) [inline spec/controllers/api/v1/open_library_controller_spec.rb]: Why do you check if the response is equal to `not_found_result` but the stubbed service is mocking the response with a wrong isbn with `empty_result`? You should have only one of them and use it accordingly.

Comment (marcegl91) [reply spec/controllers/api/v1/open_library_controller_spec.rb]: This is ok, controller response and service response are not the same thing.

**Review by marcegl91 — DISMISSED** [review]

**Review by edkajimenez — DISMISSED** [review]

**Review by marcegl91 — APPROVED** [review]

**Review by edkajimenez — APPROVED** [review]

**Pull request merged**